manager/assets/modext/widgets/media/modx.browser.js

```diff
@@ -1596,12 +1598,14 @@ Ext.extend(MODx.browser.RTE,Ext.Viewport,{
     }
 
     ,onSelect: function(data) {
-        var selNode = this.view.getSelectedNodes()[0];
-        var callback = this.config.onSelect || this.onSelectHandler;
-        var lookup = this.view.lookup;
-        var scope = this.config.scope;
+        var selNode = this.view.getSelectedNodes()[0],
+            callback = this.config.onSelect || this.onSelectHandler,
+            lookup = this.view.lookup,
+            scope = this.config.scope,
+            source = this.config.source;
         if (callback) {
             data = (selNode) ? lookup[selNode.id] : null;
+            data.source = source;
```

Comment:
Same moves as above here plus simplified the reassignment of `data` to match above (there's no need to test for truthiness of `selNode`, as a selection had to have been made for `onSelect` to be called).
```suggestion
        const selNode = this.view.getSelectedNodes()[0],
              callback = this.config.onSelect || this.onSelectHandler,
              lookup = this.view.lookup,
              scope = this.config.scope,
              source = this.config.source || MODx.config.default_media_source
        ;
        if (callback) {
            data = lookup[selNode.id];
            data.source = source ? parseInt(source) : undefined ;
```